packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_submit_ext_htlc.test.ts

```diff
@@ -0,0 +1,208 @@
+  })
+
+  afterEach(async () => {
+    await testing.icxorderbook.closeAllOpenOffers()
+  })
+
+  it('should submit EXT HTLC for a DFC buy offer', async () => {
+    // ICX order creation, make offer and submit DFC HTLC
+    const createOrder = {
+      chainTo: 'BTC',
+      ownerAddress: testing.icxorderbook.accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    const { order, createOrderTxId } = await testing.icxorderbook.createDFISellOrder(createOrder)
+
+    const makeOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.10),
+      ownerAddress: testing.icxorderbook.accountBTC
+    }
+    const { makeOfferTxId } = await testing.icxorderbook.createDFIBuyOffer(makeOffer)
+
+    const DFCHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 1440
+    }
+    await testing.icxorderbook.createDFCHTLCForDFIBuyOffer(DFCHTLC)
+
+    // submit EXT HTLC
+    const script = await providers.elliptic.script()
+    const submitEXTHTLC: ICXSubmitEXTHTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(0.10),
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      htlcScriptAddress: '13sJQ9wBWh8ssihHUgAaCmNWJbBAG5Hr9N',
+      ownerPubkey: '036494e7c9467c8c7ff3bf29e841907fb0fa24241866569944ea422479ec0e6252',
+      timeout: 24
+    }
+    const txn = await builder.icxorderbook.submitEXTHTLC(submitEXTHTLC, script)
+
+    const encoded: string = OP_CODES.OP_DEFI_TX_ICX_SUBMIT_EXT_HTLC(submitEXTHTLC).asBuffer().toString('hex')
+    const expectedRedeemScript = `6a${encoded}`
+
+    const outs = await sendTransaction(container, txn)
+    expect(outs.length).toStrictEqual(2)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[0].n).toStrictEqual(0)
+    expect(outs[0].tokenId).toStrictEqual(0)
+    expect(outs[0].scriptPubKey.asm.startsWith('OP_RETURN 4466547834')).toBeTruthy()
+    expect(outs[0].scriptPubKey.hex).toStrictEqual(expectedRedeemScript)
+    expect(outs[0].scriptPubKey.type).toStrictEqual('nulldata')
+
+    expect(outs[1].value).toEqual(expect.any(Number))
+    expect(outs[1].n).toStrictEqual(1)
+    expect(outs[1].tokenId).toStrictEqual(0)
+    expect(outs[1].scriptPubKey.type).toStrictEqual('witness_v0_keyhash')
+    expect(outs[1].scriptPubKey.reqSigs).toStrictEqual(1)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    await testing.generate(1)
+
+    // List htlc and check
+    const listHTLCOptions: ICXListHTLCOptions = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCs = await testing.rpc.icxorderbook.listHTLCs(listHTLCOptions)
+    expect(Object.keys(HTLCs).length).toBe(3) // extra entry for the warning text returned by the RPC atm.
```

Comment:
2 months ago, I am the one who refactor all toBe, toEqual to toStrictEqual in the jellyfish project. But programmers add back them

Reply:
hmm. probably coz of copied code. we might need another refactor. will update this one. for others https://github.com/DeFiCh/jellyfish/issues/613